### PR TITLE
groups -> targetFramework description is incorrect

### PR DESCRIPTION
**Pull request by kartheekp-ms**
Fixes: #1871
After going through the source code and validating my understanding with @nkolev92, I came to know the `canonical` names are used in `dependencies -> group -> targetFramework`  attribute in the `.nuspec` file. Modified example and added content in the `note` section to guide customers about the differences in the format of TFM identifier between `dependencies` group and `lib/ref` folders.

### CODE REVIEW

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[035caf5](https://github.com/NuGet/docs.microsoft.com-nuget/commits/035caf56e58c370a72daa572d7cac8f1eb61b743)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/target-frameworks.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/target-frameworks.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/nuget/reference/target-frameworks?branch=pr-en-us-1929) | [Details](#user-content-d9a31f79-7dbe9b57)
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

<a id="d9a31f79-7dbe9b57"></a>
### [docs/reference/target-frameworks.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/target-frameworks.md)
 - **Line 21**: **[Warning]** `````Invalid file link: '../reference/errors-and-warnings/nu5128'.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282250570849-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=lBBuk94bVP8XHqCYXENZCvDdOcS%2FrbeqAOu%2BvqgCXwo%3D&st=2020-02-28T22%3A46%3A36Z&se=2020-03-30T22%3A51%3A36Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[bac2ac1](https://github.com/NuGet/docs.microsoft.com-nuget/commits/bac2ac1ea8ad12bdb7efdc2f48dabdec2abf1bde)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) | [Details](#user-content-8224f5a0-d9e2fe29)
[docs/reference/target-frameworks.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/target-frameworks.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/target-frameworks?branch=pr-en-us-1929) |

<a id="8224f5a0-d9e2fe29"></a>
### [docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md)
 - **Line 381**: **[Warning]** `````Invalid file link: '../reference/errors-and-warnings/nu5128'.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282257207902-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=mCTCj1rya%2F7vw40h%2FTzDaDpTAfA6vyUsYBwVlr8Pwjw%3D&st=2020-02-28T22%3A52%3A59Z&se=2020-03-30T22%3A57%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/reference/nuspec.md [hunk 1]
```diff
@@ -368,15 +368,17 @@ The following example shows different variations of the `<group>` element:
...
         <dependency id="WebActivator" version="1.4.4" />
     </group>
 
-    <group targetFramework="sl30">
+    <group targetFramework="Silverlight4.0">
     </group>
 </dependencies>
 ```
+> [!Note]
+> The format of TFM identifier used in `assembly paths` is different when compared to the TFM identifier used in `dependency groups`. If the target frameworks declared in the dependencies group of the `nuspec` and the `lib/ref` folder do not have exact matches then `pack` command will raise [NuGet Warning NU5128](../reference/errors-and-warnings/nu5128).
```
Comment (zkat) [inline docs/reference/nuspec.md]: It would be nice to have a link here to the list of valid assembly path TFMs

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[e1f955b](https://github.com/NuGet/docs.microsoft.com-nuget/commits/e1f955b1b2a9e9c7560c5803d23e54390b4d1564)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) | [Details](#user-content-8224f5a0-5b0d3097)

<a id="8224f5a0-5b0d3097"></a>
### [docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md)
 - **Line 364**: **[Warning]** `````Invalid file link: '../reference/errors-and-warnings/nu5128'.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282302455507-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=vFDWQrW0T%2BzoBQ1EK2A%2FQsm6HV%2BpsBuUeWnBdN07Jf8%3D&st=2020-02-28T22%3A58%3A21Z&se=2020-03-30T23%3A03%3A21Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[97cf024](https://github.com/NuGet/docs.microsoft.com-nuget/commits/97cf024623242ae89441f8c560a8e86fa40a749e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282310211727-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Zbva9bu182v3WGKaSpMNSYLX665LzaWoJd2g5KNwZzM%3D&st=2020-02-28T23%3A06%3A18Z&se=2020-03-30T23%3A11%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/reference/nuspec.md [hunk 2]
```diff
@@ -360,6 +360,9 @@ The `<group>` element without a `targetFramework` attribute is used as the defau
 > [!Important]
 > The group format cannot be intermixed with a flat list.
 
+> [!Note]
+> The format of TFM identifier used in `assembly paths` is different when compared to the TFM identifier used in `dependency groups`. If the target frameworks declared in the dependencies group of the `nuspec` and the `lib/ref` folder do not have exact matches then `pack` command will raise [NuGet Warning NU5128](../reference/errors-and-warnings/nu5128.md).
```
Comment (zivkan) [inline docs/reference/nuspec.md]: Never use an acronym without it being defined earlier in the page. Change the first reference of TFM to "Target Framework Moniker (TFM), and make it a link to our target frameworks doc.

File: docs/reference/nuspec.md (hunk 1 above)
Comment (kartheekp-ms) [reply docs/reference/nuspec.md]: I have exactly the same feedback. As per my discussion with @nkolev92, it looks like TFM identifiers are not well documented and hence using [NuGet's MSBuild pack target](https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets) is an ideal solution. The solution is well documented in [NuGet Warning NU5128](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5128).

File: docs/reference/nuspec.md (hunk 2 above)
Comment (nkolev92) [reply docs/reference/nuspec.md]: I woudln't call it TFM identifier either. 

Because Target Framework Identifier exists and it might be confusing.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[82c67cf](https://github.com/NuGet/docs.microsoft.com-nuget/commits/82c67cffb12b234de182d6d01ecc007943db1172)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282319551998-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=%2BMaGoFoCvaGDtmpmpvPRJLzvjdqqoBiDRqMaARvDPkI%3D&st=2020-02-28T23%3A15%3A33Z&se=2020-03-30T23%3A20%3A33Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/reference/nuspec.md (hunk 2 above)
Comment (kartheekp-ms) [reply docs/reference/nuspec.md]: fixed in the latest commit.

File: docs/reference/nuspec.md [hunk 3]
```diff
@@ -368,12 +371,12 @@ The following example shows different variations of the `<group>` element:
...
     </group>
 
-    <group targetFramework="net40">
+    <group targetFramework=".NETFramework4.7.2">
         <dependency id="jQuery" version="1.6.2" />
         <dependency id="WebActivator" version="1.4.4" />
     </group>
 
-    <group targetFramework="sl30">
+    <group targetFramework="Silverlight4.0">
```
Comment (rrelyea) [inline docs/reference/nuspec.md]: can we update this to something that isn't silverlight... very old.  perhaps netcoreapp3.1

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[7552a7c](https://github.com/NuGet/docs.microsoft.com-nuget/commits/7552a7c3f540caa462acaa16600508fa36576bf4)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282336282915-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=2GcurfAaqO8XhsZiY%2BGqBowY59gb45CfhGy5k8NNsfI%3D&st=2020-02-28T23%3A32%3A04Z&se=2020-03-30T23%3A37%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[e2c1d5e](https://github.com/NuGet/docs.microsoft.com-nuget/commits/e2c1d5edab40c7498340a757285bd00180e75152)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282340214716-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=AdPhhlvwrbH66mtDb%2BVn846NK6d0qygW8Ke6xShCNKo%3D&st=2020-02-28T23%3A36%3A01Z&se=2020-03-30T23%3A41%3A01Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[810c896](https://github.com/NuGet/docs.microsoft.com-nuget/commits/810c896e45228c7f3a17970f57f51d339d0b60d9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/dev-kartheekp-ms-groupdependencies-1871/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1929) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002282341024412-1929%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=eUXw30199Hzdw4HO2wvvetYBxFq9YVGZysAvXUTfOFc%3D&st=2020-02-28T23%3A36%3A43Z&se=2020-03-30T23%3A41%3A43Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**